openapi-cli/src/test/java/io/ballerina/openapi/cmd/NegativeCmdTests.java

```diff
@@ -95,6 +95,61 @@ public void testFlattenWithInputOpenAPIFileParsingIssues() throws IOException {
+        String[] addArgs = {"-f", "json"};
+        Sanitize sanitize = new Sanitize(printStream, false);
+        new CommandLine(sanitize).parseArgs(addArgs);
+        sanitize.execute();
+        String output = readOutput(true);
+        Assert.assertTrue(output.contains("ERROR: an OpenAPI definition path is required to sanitize the OpenAPI " +
+                "definition"));
+    }
+
+    @Test(description = "Test with the invalid input OpenAPI file in `sanitize` sub command")
+    public void testSanitizeWithInvalidInputOpenAPIFile() throws IOException {
+        String[] args = {"-i", resourceDir + "/cmd/sanitize/openapi-1.json"};
+        Sanitize sanitize = new Sanitize(printStream, false);
+        new CommandLine(sanitize).parseArgs(args);
+        sanitize.execute();
+        String output = readOutput(true);
+        Assert.assertTrue(output.contains("ERROR: error occurred while reading the OpenAPI definition file"));
+    }
+
+    @Test(description = "Test with invalid invalid input OpenAPI file extension in `sanitize` sub command")
+    public void testSanitizeWithInvalidInputOpenAPIFileExtension() throws IOException {
+        String[] args = {"-i", resourceDir + "/cmd/sanitize/openapi.txt"};
+        Sanitize sanitize = new Sanitize(printStream, false);
+        new CommandLine(sanitize).parseArgs(args);
+        sanitize.execute();
+        String output = readOutput(true);
+        Assert.assertTrue(output.contains("ERROR: invalid input OpenAPI definition file extension. The OpenAPI " +
+                "definition file should be in YAML or JSON format"));
+    }
+
+    @Test(description = "Test with the invalid output OpenAPI file format in `sanitize` sub command")
+    public void testSanitizeWithInvalidOutputOpenAPIFileFormat() throws IOException {
+        String[] args = {"-i", resourceDir + "/cmd/sanitize/openapi.json", "-f", "txt", "-o", tmpDir.toString()};
+        Sanitize sanitize = new Sanitize(printStream, false);
+        new CommandLine(sanitize).parseArgs(args);
+        sanitize.execute();
+        String output = readOutput(true);
+        Assert.assertTrue(output.contains("WARNING: invalid output format. The output format should be either " +
+                "\"json\" or \"yaml\".Defaulting to format of the input file"));
+    }
+
+    @Test(description = "Test with the input OpenAPI file in `sanitize` sub command which has parsing issues")
+    public void testSanitizeWithInputOpenAPIFileParsingIssues() throws IOException {
+        String[] args = {"-i", resourceDir + "/cmd/sanitize/openapi_invalid.json", "-f", "txt", "-o",
+                tmpDir.toString()};
+        Sanitize sanitize = new Sanitize(printStream, false);
+        new CommandLine(sanitize).parseArgs(args);
+        sanitize.execute();
+        String output = readOutput(true);
+        Assert.assertTrue(output.contains("WARNING: invalid output format. The output format should be either " +
```

Comment:
If the user provides openapi with parser issues, are we continuing the sanitaization or terminate by providing encountered issue list?

Reply:
We are terminating if we can't get the model from the parser. If there are only warnings, we continue.